src/scattdata_header.F90

```diff
@@ -8,36 +8,53 @@ module scattdata_header
 
   implicit none
 
+
+!===============================================================================
+! JAGGED1D and JAGGED2D is a type which allows for jagged 1-D or 2-D array.
+!===============================================================================
+
+  type :: Jagged2D
+    real(8), allocatable :: data(:,:)
+  end type Jagged2D
+
+  type :: Jagged1D
+    real(8), allocatable :: data(:)
+  end type Jagged1D
+
 !===============================================================================
 ! SCATTDATA contains all the data to describe the scattering energy and
 ! angular distribution
 !===============================================================================
 
   type, abstract :: ScattData
-    ! p0 matrix on its own for sampling energy
-    real(8), allocatable :: energy(:,:) ! (Gout x Gin)
-    real(8), allocatable :: mult(:,:)   ! (Gout x Gin)
-    real(8), allocatable :: data(:,:,:) ! (Order/Nmu x Gout x Gin)
+    ! normalized p0 matrix on its own for sampling energy
+    type(Jagged1D), allocatable :: energy(:) ! (Gin % data(Gout))
+    ! nu-scatter multiplication (i.e. nu-scatt/scatt)
+    type(Jagged1D), allocatable :: mult(:)   ! (Gin % data(Gout))
+    ! Angular distribution
+    type(Jagged2D), allocatable :: dist(:)   ! (Gin % data(Order/Nmu x Gout)
```

Comment:
Might be worth putting a comment somewhere here noting that these arrays are not (necessarily) 1-indexed!